JS/Lesson34/goksel/recipe-app/src/modules/recipes/RecipeDetail/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import Card from '@mui/material/Card';
+import CardContent from '@mui/material/CardContent';
+import CardMedia from '@mui/material/CardMedia';
+import Typography from '@mui/material/Typography';
+import { CardActionArea, Stack } from '@mui/material';
+import Alert from '@mui/material/Alert';
+import { Recipe } from "../models";
+
+export const RecipeDetail = () => {
+    const { id } = useParams();
+
+    const [recipeDetail, setRecipeDetail] = useState<Recipe|null>(null);
+
+
+    useEffect(() => {
+        const URL = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${id}`
+        fetch(URL)
+            .then(res => res.json())
+            .then((data) => {
+                if (data?.meals[0]) {
+                    setRecipeDetail(data.meals[0])
+                }
+            })
+    }, [id]);
+
+    return (
+        id ?
```

Comment:
I think we should match on 'recipeDetail' instead, since if it's not present, we don't want to show recipe details

Reply:
I used recipeDetail but this time when ı click a recipe first it show me error message about 1-2 sec than showing me the details. Is it about internet connection speed?